contrib/pg_tde/documentation/docs/how-to/uninstall.md

```diff
@@ -1,59 +1,146 @@
+
+a. Clean up encrypted tables:
+
+To decrypt a table and restore it to its default storage method:
+
+```sql
+ALTER TABLE <table_name> SET ACCESS METHOD heap;
+```
+
+b. Remove the extension once all encrypted tables have been handled:
+
+```sql
+DROP EXTENSION pg_tde;
+```
+
+!!! note
+
+    If there are any encrypted objects that were not previously decrypted or deleted, this command will fail and you have to follow the steps above for these objects.
+
+## Step 2. Turn off WAL encryption
+
+If you are using WAL encryption, you need to **turn it off** before you uninstall the `pg_tde` library:
+
+a. Run:
+
+```sql
+ALTER SYSTEM SET pg_tde.wal_encrypt = off;
+```
+
+b. Restart the PostgreSQL cluster to apply the changes:
+
+- On Debian and Ubuntu:
+
+```sh
+sudo systemctl restart postgresql
+```
+
+- On RHEL and derivatives:
+
+```sh
+sudo systemctl restart postgresql-17
+```
+
+!!! note
+
+    If an error appears when restarting the PostgreSQL cluster, see [this subtopic](#troubleshooting-panic-checkpoint-not-found-on-restart).
+
+## Step 3. Uninstall the `pg_tde` shared library
 
 !!! warning
-    This process removes the extension, but does not decrypt data automatically.  Only uninstall the extension after all encrypted data **has been removed or decrypted**.
 
-To uninstall `pg_tde`, follow these steps:
+    This process removes the extension, but **does not** decrypt data automatically. Only uninstall the shared library after all encrypted data **has been removed or decrypted** and WAL encryption **has been disabled**.
 
-1. Decrypt or drop encrypted tables:
+!!! note
 
-    Before removing the extension, you must either **decrypt** or **drop** all encrypted tables:
+    Encrypted WAL pages **will not be decrypted**, so any postgres cluster needing to read them will need the `pg_tde` library loaded, and the WAL encryption keys available and in use.
 
-    - To decrypt a table, run:
+At this point, the shared library is still loaded but no longer active. To fully uninstall `pg_tde`, complete the steps below.
 
-    ```sql
-    ALTER TABLE <table name> SET ACCESS METHOD heap;
-    ```
+a. Run `SHOW shared_preload_libraries` to view the current configuration of preloaded libraries.
+
+For example:
+
+```sql
+postgres=# SHOW shared_preload_libraries;
+        shared_preload_libraries
+-----------------------------------------
+pg_stat_statements,pg_tde,auto_explain
+(1 row)
+
+postgres=#
+```
+
+b. Remove `pg_tde` from the list and apply the new setting using `ALTER SYSTEM SET shared_preload_libraries=<your list of libraries>`.
+
+For example:
+
+```sql
+postgres=# ALTER SYSTEM SET shared_preload_libraries=pg_stat_statements,auto_explain;
+ALTER SYSTEM
+postgres=#
+```
+
+!!! note
+
+    Your list of libraries will most likely be different than the above example.
+    
+    If `pg_tde` is the only shared library in the list, and it was set via `postgresql.conf`:
```

Comment:
List is not rendered

Reply:
<img width="724" height="185" alt="Screenshot 2025-08-01 at 16 12 35" src="https://github.com/user-attachments/assets/bfbeeefe-2c90-49ba-a862-a7938b57cbbb" />
Then you should rework it. since now it's clearly a list and it's broken